sdks/python/apache_beam/transforms/cy_combiners.py

```diff
@@ -312,44 +310,53 @@ class AllCombineFn(AccumulatorCombineFn):
   _accumulator_type = AllAccumulator
 
 
-MAX_LONG_10_FOR_LEADING_ZEROS = [19, 18, 18, 18, 18, 17, 17, 17, 16, 16, 16, 15,
-                                 15, 15, 15, 14, 14, 14, 13, 13, 13, 12, 12, 12,
-                                 12, 11, 11, 11, 10, 10, 10, 9, 9, 9, 9, 8, 8,
-                                 8, 7, 7, 7, 6, 6, 6, 6, 5, 5, 5, 4, 4, 4, 3, 3,
-                                 3, 3, 2, 2, 2, 1, 1, 1, 0, 0, 0]
-
-LONG_SIZE = 64
-
-
-def compare_to(x, y):
-  """return the sign bit of x-y"""
-  if x < y:
-    return 1
-  return 0
+POWER_TEN = [10e-1, 10e0, 10e1, 10e2, 10e3, 10e4, 10e5,
+             10e6, 10e7, 10e8, 10e9, 10e10, 10e11,
+             10e12, 10e13, 10e14, 10e15, 10e16, 10e17,
+             10e18]
 
 
 def get_log10_round_to_floor(element):
-  number_of_leading_zeros = LONG_SIZE - element.bit_length()
-  y = MAX_LONG_10_FOR_LEADING_ZEROS[number_of_leading_zeros]
-  return y - compare_to(element, math.pow(10, y))
+  power = 0
+  while element >= POWER_TEN[power]:
```

Comment:
Do we assume that element would be a signed integer? sys.maxint would be >= 10e18, and in that case would this loop use an out of bounds index?

Reply:
I think sys.maxint < power(10,19)